integration-tests/tests/billing-projects.js

```diff
@@ -131,17 +174,116 @@ const testBillingSpendReportFn = withUserToken(async ({ page, testUrl, token })
   await billingPage.assertChartValue(3, 'Third Most Expensive Workspace', 'Storage', '$0.00')
 
   // Change the returned mock cost to mimic different date ranges.
-  await setAjaxMockValues(page, billingProjectName, '1110.17', 20)
+  await setAjaxMockValues(page, ownedBillingProjectName, notOwnedBillingProjectName, '1110.17', 20)
   await billingPage.setSpendReportDays(90)
   await billingPage.assertText('Total spend$1,110.17')
   // Check that title updated to reflect truncation.
   await billingPage.assertText('Top 10 Spending Workspaces')
   await billingPage.assertChartValue(10, 'Extra Inexpensive Workspace', 'Compute', '$0.01')
+
+  // Select a billing project that is not owned by the user
+  await billingPage.visit()
+  await billingPage.selectProject(notOwnedBillingProjectName)
+
+  //Check that the Spend report tab is not visible on this page
+  await assertTextNotFound(billingPage, 'Spend report')
 })
 
 const testBillingSpendReport = {
   name: 'billing-spend-report',
   fn: testBillingSpendReportFn
 }
 
-module.exports = { testBillingSpendReport }
+const testBillingWorkspacesFn = withUserToken(async ({ page, testUrl, token }) => {
+  // Sign in. This portion of the test is not mocked.
+  await page.goto(testUrl)
+  await signIntoTerra(page, token)
+  await dismissNotifications(page)
+
+  // Interact with the Billing Page via mocked AJAX responses.
+  const ownedBillingProjectName = 'OwnedBillingProject'
+  const notOwnedBillingProjectName = 'NotOwnedBillingProject'
+  await setAjaxMockValues(page, ownedBillingProjectName, notOwnedBillingProjectName, '0')
+
+  const billingPage = billingProjectsPage(page, testUrl)
+
+  // Select a billing project that is owned by the user
+  await billingPage.visit()
+  await billingPage.selectProject(ownedBillingProjectName)
+
+  // Check that the Workspaces tab is visible on this page
+  await billingPage.assertText('Workspaces')
+  await billingPage.assertText('Name')
+  await billingPage.assertText('Created By')
+  await billingPage.assertText('Last Modified')
+
+  // Select a billing project that is not owned by the user
+  await billingPage.visit()
+  await billingPage.selectProject(notOwnedBillingProjectName)
+
+  // Check that the Workspaces tab is visible on this page
+  await billingPage.assertText('Workspaces')
+  await billingPage.assertText('Name')
+  await billingPage.assertText('Created By')
+  await billingPage.assertText('Last Modified')
+})
+
+const testBillingWorkspaces = {
+  name: 'billing-workspaces',
+  fn: testBillingWorkspacesFn
+}
+
+const testBillingMembersFn = withUserToken(async ({ page, testUrl, token }) => {
+  // Sign in. This portion of the test is not mocked.
+  await page.goto(testUrl)
```

Comment:
I would suggest a helper method for lines 238-247 that is shared between the 3 tests.

Reply:
Test setup extracted in a9c945847d892cfbcebc852875507738dead4f59